File: 力扣/学习计划/剑指 Offer I/014_从上到下打印二叉树 I/main.java
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

import common.TreeNode;

/**
 * https://leetcode-cn.com/problems/cong-shang-dao-xia-da-yin-er-cha-shu-lcof/
 */
class Solution {
    public int[] levelOrder(TreeNode root) {
        if (root == null) return null;
        
        List<Integer> resList = new ArrayList<>();
        Queue<TreeNode> queue = new LinkedList<>() {{ offer(root); }};
        while (!queue.isEmpty()) {
            TreeNode node = queue.poll();
            resList.add(node.val);

            if (node.left != null)
                queue.offer(node.left);
            if (node.right != null)
                queue.offer(node.right);
        }
        return resList.stream().mapToInt(e -> e.intValue()).toArray();
    }
}

File: 力扣/学习计划/剑指 Offer I/014_从上到下打印二叉树 I/main.go
package main

func levelOrder(root *TreeNode) []int {
	if root == nil {
		return []int{}
	}

	var res = make([]int, 0)

	queue := make([]*TreeNode, 0)
	queue = append(queue, root)
	for len(queue) != 0 {
		node := queue[0]
		queue = queue[1:] // 模拟队列出队

		res = append(res, node.Val)
		if node.Left != nil {
			queue = append(queue, node.Left)
		}
		if node.Right != nil {
			queue = append(queue, node.Right)
		}
	}
	return res
}

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}
